### PR TITLE
Support --temp-build also in `dub build`

### PR DESCRIPTION
**Pull request by gizmomogwai**
This allows for complete out of git builds. The link command now also
prints the resulting filename (although that might be too much for
normal usage).

### CODE REVIEW

File: source/dub/commandline.d [hunk 1]
```diff
@@ -678,7 +678,6 @@ class CommandArgs {
 	{
 		foreach (ref arg; m_recognizedArgs)
 			if (names == arg.names) {
-				assert(help_text is null);
```
Comment (WebFreak001) [inline source/dub/commandline.d:681]: don't remove this assert, it checks if there are duplicate recognized args, and if there are the duplicates must not change the help text.

Instead you want to remove the `temp-build` getopt from RunCommand, as you now added it in BuildCommand, which RunCommand inherits from

**Review by WebFreak001 — CHANGES_REQUESTED** [review]

File: source/dub/commandline.d (hunk 1 above)
Comment (gizmomogwai) [reply source/dub/commandline.d:681]: thanks for the explanation. i reworked the commits and added a perhaps clearer message to the error.

File: source/dub/commandline.d [hunk 2]
```diff
@@ -678,7 +678,7 @@ class CommandArgs {
...
+				enforce(help_text is null, format!("duplicated argument '%s' must not change helptext, consider to remove the duplication")(names));
```
Comment (WebFreak001) [inline source/dub/commandline.d:681]: I think this is a good addition now, but it should stay an assert (it's a programming bug, not a runtime exception that may occur)

Comment (gizmomogwai) [reply source/dub/commandline.d:681]: true ...

Comment (WebFreak001) [reply source/dub/commandline.d:681]:
```suggestion
				assert(help_text is null, format!("duplicated argument '%s' must not change helptext, consider to remove the duplication")(names));
```

File: source/dub/generators/build.d [hunk 3]
```diff
@@ -579,7 +579,7 @@ class BuildGenerator : ProjectGenerator {
 			settings.compiler.invoke(buildsettings, settings.platform, settings.compileCallback);
 
 			if (generate_binary) {
-				logInfo("Linking", Color.light_green, "%s", buildsettings.targetName.color(Mode.bold));
+				logInfo("Linking", Color.light_green, "%s => %s", buildsettings.targetName.color(Mode.bold), buildsettings.getTargetPath(settings));
```
Comment (WebFreak001) [inline source/dub/generators/build.d:582]: what do others think about this CLI output change? @Geod24 @thewilsonator ?

The output path is for example `/tmp/.dub/build/issue2051-test-application-unittest-linux.posix-x86_64-dmd_v2.100.0-0AC041139053DBA3B189AED31C7F8A27221CA401DEE41BA7B05C3F5F74C4D4BE/issue2051-test-application`, possibly even longer for longer project folder names in local temp cache.

I don't think this is really useful most of the time, should we only show this when having the --temp-build option used?

Comment (gizmomogwai) [reply source/dub/generators/build.d:582]: yes ... that would be my preference as well! normally its not really needed!

**Review by WebFreak001 — APPROVED** [review]

**Pull request merged**